javers-core/src/test/groovy/org/javers/core/JaversRepositoryE2ETest.groovy

```diff
@@ -689,4 +692,47 @@ class JaversRepositoryE2ETest extends Specification {
             byInstanceId(1, SnapshotEntity).withVersion(5).build()
         ]
     }
+
+    def "should retrieve snapshots with specified descriptors"() {
+        given:
+        (1..10).each {
+            javers.commit("author", new SnapshotEntity(id: 1, intProperty: it))
+            javers.commit("author", new SnapshotEntity(id: 2, intProperty: it))
+            javers.commit("author", new SnapshotEntity(id: 3, intProperty: it))
+        }
+
+        def snapshotDescriptors = [
+            new SnapshotDescriptor(javers.idBuilder().instanceId(new SnapshotEntity(id: 1)), 3),
```

Comment:
bardziej spójnie z resztą JQL będzie jeśli SnapshotDescriptor będzie miał `GlobalIdDTO` a nie `GlobalId` (tak jak np w IdFilter). Z GlobalId jest taki problem, że jest to ciężki obiekt domenowy javersa i trudno się go tworzy.
Do dtos'ów są wygodne buildery, np `InstanceIdDTO.instanceId(localId, entityClass)`